### DISCUSSION TITLE
No DpId related info in TuyaReceived

### DISCUSSION THREAD

**Discussion by leeza1971** [arendst/Tasmota#15933] [Q&A] [answered]
It is a Tuya 6 gang wall switch.
I replace the WIFI module with ESP-12F flashed with Tasmota, and configured the module as TuyaMCU device (Tasmota ver. 12.0.2).
ESP-12F is communicating with TuyaMCU but the message "TuyaReceived" does not contain any info related to DpId.

Here is the log (with Weblog 4).

-------------------------------------------------------------------------------------------------
10:17:12.434 TYA: Send "55aa00000000ff"
10:17:12.437 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
10:17:12.439 TYA: Heartbeat
10:17:23.428 TYA: Send "55aa00000000ff"
10:17:23.431 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
10:17:23.432 TYA: Heartbeat
10:17:27.017 WIF: Checking connection...
10:17:34.433 TYA: Send "55aa00000000ff"
10:17:34.436 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
10:17:34.438 TYA: Heartbeat
10:17:45.425 TYA: Send "55aa00000000ff"
10:17:45.428 {"TuyaReceived":{"Data":"55AA030000010104","Cmnd":0,"CmndData":"01"}}
-------------------------------------------------------------------------------------------------

As far as I know, the message "TuyaReceived" should contain info such as DpType and DpIdData as shown in an example below (which I got off the Internet).

-------------------------------------------------------------------------------------------------
{"TuyaReceived":{"Data":"55AA0107000501010001000F","Cmnd":7,"CmndData":"0101000100","DpType1Id1":0,"1":{"DpId":1,"DpIdType":1,"DpIdData":"00"}}}
TYA: fnId=11 is set for dpId=1
TYA: RX Relay-1 --> MCU State: Off Current State:Off
-------------------------------------------------------------------------------------------------

Since the message "TuyaReceived" does not contain any info on DpId, I manually issued the command "backlog TuyaMCU 11,1; TuyaMCU 12,2; TuyaMCU 13,3; TuyaMCU 14,4; TuyaMCU 15,5; TuyaMCU 16,6".

While the WebUI shows six relays after the command was issued, the states of the relays are not updated when I use the touch switches of the wall switch as the message "TuyaReceived" does not contain any info for updating the state of the relay,

In addition, the WebUI with six relays after the command was issued looks somewhat different. It does not have "ON" "OFF". It just have "0" and "1" .

![image](https://user-images.githubusercontent.com/43492435/177520321-a1fc9306-a20d-4e24-aea2-5a694ca82080.png)

Any help would be appreciated.

**Comment by sfromis** [comment]
What those `TuyaReceived` messages have is only the heartbeat. It was not dead, but had nothing to say.

**Reply by leeza1971** [reply]
> What those `TuyaReceived` messages have is only the heartbeat. It was not dead, but had nothing to say.

Why is the MCU transmitting only heartbeat? Why is the MCU not transmitting DpId related information?

Do I need to issue a specific command to receive DpId related information?

I have IR remote which is a Tuya device. I flashed IR remote with Tasmota and configured the module as TuyaMCU and it started working without any command (except weblog 4 to see the log).

I have four of these wall switches and none of them are working...

**Comment by barbudor** [comment, marked as answer]
> While the WebUI shows six relays after the command was issued, the states of the relays are not updated when I use the touch switches of the wall switch as the message "TuyaReceived" does not contain any info for updating the state of the relay,

Are the relay activated when you press the buttons ?
Probably yes as the touch buttons are handled directly by the TuyaMCU
Do you see any "Tuya Received logs" when you press the buttons ?
If not, it means that the TuyaMCU do not bother to let the ESP knows about that which makes the DpId almost impossible to identify

> In addition, the WebUI with six relays after the command was issued looks somewhat different. It does not have "ON" "OFF". It just have "0" and "1" .

This is expected to have all those buttons on a single line.
Occur with more than 4 buttons

**Reply by leeza1971** [reply]
The relays are activated when the touch buttons are pressed.
When I press a button, it looks like MCU is not sending any status messages according to the log.
This is a very strange behavior of the MCU.

I tested one of the four switches which is not flashed with Tasmota using Tuya app.

It works without any problem. I can see the status of the relays on the Tuya app, which means that MCU is somehow sending the status of relay to the WIFI module.

When I replace the WIFI module with ESP-12F flashed with Tasmota, it just does not work.

**Reply by barbudor** [reply]
I you happen to have a logic analyser, you could be able to spy the serial between MCU and esp with original firmware
As far as you can do that WITHOUT CONNECTING TO Mains for abvious security reasons

Otherwise here ends my knowledge on Tuya

**Reply by sfromis** [reply]
While I know no details (I avoid TuyaMCU devices whenever possible), my general impression is that you may have to send some sort of initialization to the TuyaMCU to tell it what it is supposed to send. Thus, it could indeed be relevant to listen to the communication happening with factory firmware.
